### PR TITLE
Remove use of multiple structured suffixes

### PR DESCRIPTION
**Pull request by selfissued**
Aligns with https://github.com/w3c/vc-data-model/pull/1478/

***
<a href="https://pr-preview.s3.amazonaws.com/selfissued/vc-jose-cose/pull/279.html" title="Last updated on Jun 25, 2024, 5:09 PM UTC (802c8a9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/279/302fd95...selfissued:802c8a9.html" title="Last updated on Jun 25, 2024, 5:09 PM UTC (802c8a9)">Diff</a>

### CODE REVIEW

**Review by decentralgabe — APPROVED** [review]
great work! I will make sure to update the examples after this goes in

File: index.html [hunk 1]
```diff
@@ -1429,13 +1422,13 @@ <h2 id="securing-verifiable-credentials">Securing Verifiable Credentials</h2>
...
-            <code>application/vp+ld+json+sd-jwt</code>,
+            <code>application/vc</code> and
+            <code>application/vp</code>,
+            <code>application/vc+sd-jwt</code> and
+            <code>application/vp+sd-jwt</code>,
             as well as
-            <code>application/vc+ld+json+cose</code> and
-            <code>application/vp+ld+json+cose</code>.
+            <code>application/vc+cose</code> and
+            <code>application/vp+cose</code>.
```
Comment (OR13) [inline index.html:1431]: vc+jwt and vp+jwt are missing in this list

File: index.html [hunk 2]
```diff
@@ -1573,10 +1566,10 @@ <h2 id="vp-ld-json-jwt"><code>application/vp+ld+json+jwt</code></h2>
         </section>
 
         <section id="vc-ld-sd-jwt-media-type">
-            <h2 id="vc-ld-json-sd-jwt"><code>application/vc+ld+json+sd-jwt</code></h2>
+            <h2 id="vc-ld-json-sd-jwt"><code>application/vc+sd-jwt</code></h2>
             <p>
                 This specification registers the
-                <code>application/vc+ld+json+sd-jwt</code> Media Type specifically for
+                <code>application/vc+sd-jwt</code> Media Type specifically for
```
Comment (OR13) [inline index.html:1572]: This registration is already requested in https://datatracker.ietf.org/doc/html/draft-ietf-oauth-sd-jwt-vc-03#name-application-vcsd-jwt

**Review by OR13 — COMMENTED** [review]
@awoie @bc-pi @danielfett 

Hello, authors of SD-JWT-based Verifiable Credentials (SD-JWT VC)

It seems the W3C VCWG has decided to stop using multiple suffixes, and as such is now requesting registration of:

application/vc+sd-jwt to convey a "W3C JSON-LD JWT claim set as an SD-JWT payload".

This makes JSON-LD support no longer detectable by looking at the media type.

The examples you have in https://datatracker.ietf.org/doc/html/draft-ietf-oauth-sd-jwt-vc-03#name-media-type

Are not conformant with the requirements for a "JSON-LD claimset", there is no `@context`, and other required fields such as `issuer` and `credentialSubject`, and `type` are not present.

I can imagine you may have comments on this pull request.

As a general comment, it should be possible to know what fields will be required or optional by looking at the media type, and there should not be 2 non interoperable expressions, that both claim to be the same media type.

One path forward might be to add explanatory text to whichever document does register the `application/vc+sd-jwt` media type, that the claimset is always a JWT claimset, and might contain JSON-LD keywords, and the `@context` member of the JWT claimset can be used to distinguish a JWT claimset from a JSON-LD one.

**Comment by bc-pi** [comment]
> I can imagine you may have comments on this pull request.

Yes, please consider using something other than `application/vc+sd-jwt` for identifying an SD-JWT with a payload conforming to the Verifiable Credential Data Model. 

Perhaps one of:
`application/vcdm+sd-jwt`
`application/vcdm2+sd-jwt`
`application/w3c-vc+sd-jwt`
`application/vc-ld+sd-jwt`
`application/vc-json-ld+sd-jwt`
`application/vc-ld-json+sd-jwt`

**Review by bc-pi — CHANGES_REQUESTED** [review]
https://github.com/w3c/vc-jose-cose/pull/279#issuecomment-2181518331

**Comment by selfissued** [comment]
My initial reaction to the suggestion to change only `application/vc+sd-jwt` but not any of the other related proposed media types such as `application/vp+sd-jwt`, `application/vc+jwt`, `application/vc+cose` is that that's probably a non-starter with the VCWG because of the obviously inconsistencies in naming.

As I'd written to the OAuth WG, it’s my hope that the registrations of `application/vc+sd-jwt` and `application/vp+sd-jwt` will be able to be done in a way that works for both VC-JOSE-COSE and SD-JWT-VC.  I'd like people's input on that.

If that ends up being infeasible, then yes, we'll have to come up a way to deconflict the names used by VC-JOSE-COSE and SD-JWT-VC.  Thanks all for thinking about this.

File: index.html (hunk 2 above)
Comment (bc-pi) [reply index.html:1572]: Pedantically speaking the request has not yet been made but there is intent to do so. https://mailarchive.ietf.org/arch/msg/media-types/2VdBm8c-GVcYf3GrWGyQSkJ-NVM/

**Comment by selfissued** [comment]
Understood.  @bc-pi, can you please file an issue for vc-data-model asking to change `application/vc` to `application/vc-ld` and `application/vp` to `application/vp-ld`?  That change would set the stage for the VCWG to be able to consistently use names like `application/vc-ld+sd-jwt`, etc.  Thanks.

**Comment by OR13** [comment]
In case folks wish to provide comments or review to OAuth or Spice, see

https://mailarchive.ietf.org/arch/msg/oauth/2qfLP2RFEehzUxrdmiXTAEqk7OQ/

https://github.com/oauth-wg/oauth-sd-jwt-vc

https://github.com/mesur-io/ietf-misc/blob/main/editor_drafts/draft-prorock-spice-cose-sd-cwt.md?plain=1 ( hopefully sd-cwt draft will move elsewhere soon ).

**Comment by bc-pi** [comment]
> Understood. @bc-pi, can you please file an issue for vc-data-model asking to change `application/vc` to `application/vc-ld` and `application/vp` to `application/vp-ld`? That change would set the stage for the VCWG to be able to consistently use names like `application/vc-ld+sd-jwt`, etc. Thanks.

yup, https://github.com/w3c/vc-data-model/issues/1509

**Review by OR13 — APPROVED** [review]

**Review by decentralgabe — APPROVED** [review]
re-approving. thanks @selfissued

File: index.html [hunk 3]
```diff
@@ -1429,13 +1421,15 @@ <h2 id="securing-verifiable-credentials">Securing Verifiable Credentials</h2>
         </p>
         <p>The normative statements in <a data-cite="VC-DATA-MODEL-2.0#securing-mechanisms">Securing
             Mechanisms</a> apply to securing
-            <code>application/vc+ld+json</code> and
-            <code>application/vp+ld+json</code>,
-            <code>application/vc+ld+json+sd-jwt</code> and
-            <code>application/vp+ld+json+sd-jwt</code>,
+            <code>application/vc</code> and
+            <code>application/vp</code>,
```
Comment (bc-pi) [inline index.html:1425]: I've honestly never really understood this section but it feels like it should only make whatever statement this is making for the media types which this document defines. 
```suggestion
            <code>application/vc</code> and
            <code>application/vp</code>,
```
```suggestion
```

**Review by bc-pi — APPROVED** [review]
one nit/suggestion but otherwise approve. Thanks @selfissued

**Comment by decentralgabe** [comment]
Editorial approval, changes addressed, open for over a week. Merging.

**Pull request merged**